assets/tests/consolidated_1_1_tests.yaml

```diff
@@ -5295,3 +5295,294 @@ tests:
+            expectation: []
+  - name: userset_defines_itself_4
+    stages:
+      - model: |
+          model
+            schema 1.1
+
+          type user  
+
+          type folder
+            relations
+              define viewer: [user]
+
+          type document
+            relations
+              define parent: [folder]
+              define viewer: viewer from parent
+        tuples:
+          - user: folder:x
+            relation: parent
+            object: document:1
+        checkAssertions:
+        - tuple:
+            user: folder:x#viewer
+            relation: viewer
+            object: document:1
+          expectation: true
+        - tuple:
+            user: document:1#viewer
+            relation: viewer
+            object: document:1
+          expectation: true
+        listObjectsAssertions:
+          - request:
+              user: document:1#viewer
+              type: document
+              relation: viewer
+            expectation: ["document:1"]
+          - request:
+              user: folder:x#viewer
+              type: document
+              relation: viewer
+            expectation: ["document:1"]
+  - name: userset_defines_itself_5
+    stages:
+      - model: |
+          model
+            schema 1.1
+
+          type user
+
+          type document
+            relations
+              define allowed: [user]
+              define viewer: [user] and allowed
+        checkAssertions:
+        - tuple:
+            user: document:1#viewer
+            relation: allowed
+            object: document:1
+          expectation: false
+        - tuple:
+            user: document:1#allowed
+            relation: viewer
+            object: document:1
+          expectation: false
+        - tuple:
+            user: document:1#viewer
+            relation: viewer
+            object: document:1
+          expectation: true
+        listObjectsAssertions:
+          - request:
+              user: document:1#allowed
+              type: document
+              relation: viewer
+            expectation: []
+          - request:
+              user: document:1#viewer
+              type: document
+              relation: allowed
+            expectation: []
+  - name: userset_defines_itself_6
+    stages:
+      - model: |
+          model
+            schema 1.1
+
+          type user
+
+          type document
+            relations
+              define restricted: [user]
+              define viewer: [user] but not restricted
+        checkAssertions:
+        - tuple:
+            user: document:1#viewer
+            relation: restricted
+            object: document:1
+          expectation: false
+        - tuple:
+            user: document:1#restricted
+            relation: viewer
+            object: document:1
+          expectation: false
+        - tuple:
+            user: document:1#viewer
+            relation: viewer
+            object: document:1
+          expectation: true
```

Comment:
missing the equivalent ListObejcts assertion

Reply:
I have the test case `Check(x#y@x#y)` tested in various other places, but I'll add it here for good measure too.